### ISSUE TITLE
Deleting a layer makes all layers black

### ISSUE THREAD

**Issue by brannondorsey** [kibio/kibio#31]
![screen shot 2015-03-13 at 10 33 44 pm](https://cloud.githubusercontent.com/assets/3193533/6650143/1ce73bbc-c9d1-11e4-82ed-fadf01eee01c.png)

**Comment by brannondorsey** [comment]
As soon as a layer is deleted all others go black and this is spewed continuously:
```
[ error ] ofxiOSVideoPlayer: initTextureCache(): error creating texture cache from image -6661
```
It looks like it is coming from [here](https://github.com/kibio/kibio/blob/2bbc54f8f53c96be5810b33ba3b68830d1f457b7/src/Project.cpp#L282-L284) in `Project.cpp`:

```cpp
_layers.erase(std::find(_layers.begin(), _layers.end(), layer));
```

**Comment by brannondorsey** [comment]
From (`C++ `vector::erase` documentation](http://www.cplusplus.com/reference/vector/vector/erase/): "Because vectors use an array as their underlying storage, erasing elements in positions other than the vector end causes the container to relocate all the elements after the segment erased to their new positions."

I bet we have passed pointers to ofVideoPlayer those pointers become invalid when erased. Does this sound plausible? I will look into this.

**Comment by brannondorsey** [comment]
The `[ error ] ofxiOSVideoPlayer` is coming from ofxAVFoundationVideoPlayer [here](https://github.com/julapy/ofxAVFoundationVideoPlayer/blob/master/src/ofxAVFoundationVideoPlayer.mm#L414).

**Comment by bakercp** [comment]
This appears to be fixed with 0.2.2.
